agency/src/services/schema/schema.class.ts

```diff
@@ -65,33 +65,120 @@ export class Schema implements ServiceSwaggerAddon {
 
       if (isNewSchema) {
         // post schema on ledger
-        const schemaResponse = (await this.app.service('aries-agent').create({
+        const schemaAuthorTxn = await this.app.service('aries-agent').create({
           service: ServiceType.Schema,
           action: SchemaServiceAction.Create,
           token: params.profile.wallet.token,
           data: {
             schema_name: schema.schema_name,
             schema_version: schema.schema_version,
             attributes: schema.attributes,
+            conn_id: params?.profile?.endorser?.connection_id || '',
           } as AriesSchemaServiceRequest,
-        } as AriesAgentData)) as AriesSchema;
-        schema.schema_id = schemaResponse.schema_id || schemaResponse.schema.id;
+        } as AriesAgentData);
+
+        if (!schemaAuthorTxn) {
+          throw new ErrorEvent('Invalid Schema author transaction returned');
+        }
```

Comment:
Since these errors are breaking the workflow, could we surface the error code/message coming from the endorser so that the `AriesAgentError` handler that will catch and return the error to the API consumer will have a more descriptive message/context? Does that make sense?

Same below where we are throwing errors, if applicable.

Reply:
Should we generate an interface for this?

Reply:
Also would the AriesAgentError not bubble up to be caught by the method handler already?